test/ufuzz.js

```diff
@@ -582,6 +582,14 @@ function _createExpression(recurmax, noComma, stmtDepth, canThrow) {
       case 30:
         return '(' + createObjectLiteral(recurmax, COMMA_OK, stmtDepth, canThrow) + '.' +
             SAFE_KEYS[rng(SAFE_KEYS.length)] + ") ";
+      case 31:
+        var name = VAR_NAMES[rng(VAR_NAMES.length)];
+        return '(' + name + ' && ' + name + '[' +
+            createExpression(recurmax, COMMA_OK, stmtDepth, canThrow) + "]) ";
+      case 32:
+        var name = VAR_NAMES[rng(VAR_NAMES.length)];
```

Comment:
`x.prop = expr` alone would, yes - I always generate `x && x.prop = expr` with https://github.com/mishoo/UglifyJS2/pull/1783/commits/a5244b5974c4d59d90ded37c5b937b31b36fa5fb though.

But good point - may be if `canThrow` is true, I can skip the `x &&` bit.